t/t2200-job-ingest.t

```diff
@@ -115,7 +115,7 @@ test_expect_success 'job-ingest: jobspec stored accurately in KVS' '
 '
 
 test_expect_success 'job-ingest: job announced to job manager' '
-	jobid=$(flux job submit --priority=10 basic.json) &&
+	jobid=$(flux job submit --priority=10 basic.json | flux job id) &&
```

Comment:
One potential gotcha with this pattern is that the pipeline exit code is that of the last command, so a failure of `flux job submit` might not be caught.  Would it make sense to `set -o pipefail` at the top of the test script?

Reply:
Good catch! I was assuming that if `flux job submit` fails to emit a valid jobid on stdout, then `flux job id` will fail, but sadly that is not the case.

Unfortunately `pipefail` is a bashism, so we can't rely on that in our testsuite.

Well back to the drawing board.

Reply:
I guess the only sane way to do this is to transform the jobid after `flux job submit`, `flux mini submit` or `flux mini run -v` captures the jobid.

Until this is addressed I've set WIP on this PR.

Reply:
Would it be easier to have flux job id return an error if its stdin is empty?

Reply:
Yes, I wasn't sure if that would be acceptable. Could one of the pipe commands "fail" but still emit a valid jobid, so returning the exit code of the last command in the pipe would still be a valid concern?

Reply:
`flux job submit` will succeed if it prints the jobid, and all error messages go to stderr, so at least in that case it should be safe.

Reply:
Ok, I've pushed a commit that changes the behavior of `flux job id` to exit with failure if no jobids are processed. This should cause the idiom `flux job submit | flux job id` to fail when `flux job submit` fails, as well as `flux mini submit | flux job id` etc.